pkg/controller/sdk/reconciler.go

```diff
@@ -65,11 +65,12 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return reconcile.Result{}, err
 	}
 
-	original := obj.DeepCopyObject()
+	// Don't modify the cache's copy
+	obj := original.DeepCopyObject()
```

Comment:
Optional: Can we call this something like `reconciledObj` or `updated` to make clear that this holds the updated state?